bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java

```diff
@@ -0,0 +1,925 @@
+        QbusBridgeHandler handler = bridgeCallBack;
+        ctdConnected = false;
+
+        if (qEventsRunning) {
+            throw new IOException("Previous listening thread is still active.");
+        }
+
+        if (handler == null) {
+            throw new IOException("No Bridge handler initialised.");
+        }
+
+        InetAddress addr = InetAddress.getByName(handler.getAddress());
+        Integer port = handler.getPort();
+
+        if (port != null) {
+            Socket socket = new Socket(addr, port);
+            qSocket = socket;
+            qOut = new PrintWriter(socket.getOutputStream(), true);
+            qIn = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            logger.debug("Connected via local port {} from thread {}", socket.getLocalPort(),
+                    Thread.currentThread().getId());
+        } else {
+            return;
+        }
+
+        setSN();
+        getSN();
+
+        // Connect to Qbus server
+        connect();
+
+        // If Qbus Client is connected then initialize and start listening for incoming commands, else put Bridge
+        // offline
+        if (ctdConnected) {
+            initialize();
+            threadExecutor.execute(() -> {
+                try {
+                    qEvents();
+                } catch (IOException e) {
+                    logger.debug("Could not start thread {} ", e.getMessage());
+                }
+            });
+        } else {
+            handler.bridgeOffline("No communication with Qbus client");
+        }
+    }
+
+    /**
+     * Cleanup socket when the communication with Qbus Server is closed.
+     *
+     * @throws IOException
+     *
+     */
+    public synchronized void stopCommunication() {
+        listenerStopped = true;
+
+        Socket socket = qSocket;
+
+        if (socket != null) {
+            try {
+                socket.close();
+            } catch (IOException ignore) {
+                // ignore IO Error when trying to close the socket if the intention is to close it anyway
+            }
+        }
+
+        qSocket = null;
+        if (qOut != null) {
+            qOut.close();
+        }
+
+        try {
+            if (qIn != null) {
+                qIn.close();
+            }
+        } catch (IOException e) {
+            logger.debug("Error on closing reader {} ", e.getMessage());
+        }
+        ctdConnected = false;
+        logger.debug("Communication stopped from thread {}", Thread.currentThread().getId());
+    }
+
+    /**
+     * Close and restart communication with Qbus Server.
+     *
+     * @throws InterruptedException
+     */
+    public synchronized void restartCommunication() {
+        stopCommunication();
+
+        logger.debug("Qbus: restart communication from thread {}", Thread.currentThread().getId());
+
+        try {
+            startCommunication();
+        } catch (InterruptedException | IOException e) {
+        }
```

Comment:
The `InterruptedException` shouldn't be cought here, but as high as possible in the program flow to exit the program as directly as possible. You can add `throws InterruptedException` to the method signature to do so.

Nevertheless, the `IOException` should be logged here.